python310Packages.insightface: init at 0.7.3

oddlama opened the pull request:
###### Description of changes

Adds https://github.com/deepinsight/insightface, a state-of-the-art 2D and 3D face analysis library.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

misuzu commented on pkgs/development/python-modules/insightface/default.nix:
```diff
@@ -0,0 +1,59 @@
+
+  meta = with lib; {
+    description = "State-of-the-art 2D and 3D Face Analysis Project";
+    homepage = "https://github.com/insightface/insightface";
```
```suggestion
    homepage = "https://github.com/deepinsight/insightface";
```

oddlama replied:
Thanks for noticing!

oddlama commented:
Rebased on master

mweinelt commented on pkgs/development/python-modules/insightface/default.nix:
```diff
@@ -0,0 +1,59 @@
+    prettytable
+  ];
+
+  doCheck = false; # Upstream has no tests
```
In that case it would be good to have `pythonImportsCheck`. The following imports are mentioned in their [README](https://github.com/deepinsight/insightface/tree/master/python-package).

```suggestion
	pythonImportsCheck = [
	  "insightface"
	  "insightface.app"
	  "insightface.data"
	];

  doCheck = false; # Upstream has no tests
```

oddlama replied:
Done!

mweinelt commented:
The CLI apparently expects the `onnxruntime` package.

```
$ ./result/bin/insightface-cli
Traceback (most recent call last):
  File "/nix/store/l2j9xhb2zbl21id24xy5kgly0vb7mx74-python3.11-insightface-0.7.3/lib/python3.11/site-packages/insightface/__init__.py", line 8, in <module>
    import onnxruntime
ModuleNotFoundError: No module named 'onnxruntime'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/nix/store/l2j9xhb2zbl21id24xy5kgly0vb7mx74-python3.11-insightface-0.7.3/bin/.insightface-cli-wrapped", line 6, in <module>
    from insightface.commands.insightface_cli import main
  File "/nix/store/l2j9xhb2zbl21id24xy5kgly0vb7mx74-python3.11-insightface-0.7.3/lib/python3.11/site-packages/insightface/__init__.py", line 10, in <module>
    raise ImportError(
ImportError: Unable to import dependency onnxruntime.

oddlama commented:
Added small compatibility fix for mxnet (to support numpy >= 1.24), and addressed review comments.

> The CLI apparently expects the `onnxruntime` package. Maybe we can add `testers.testVersion` to run the CLI as a further test.

Good idea. I've now added this test but since there is no "--version" for insightface, I've set `version = "help"` to verify at least the help output.

oddlama commented:
(Fixed accidentally deleted character 😅, package builds and runs now)

mweinelt commented:
mxnet fails to build on darwin.

> error: unknown warning option '-Werror=maybe-uninitialized'; did you mean '-Werror=uninitialized'? [-Werror,-Wunknown-warning-option]

Can you try the following? Copied from `pkgs/tools/networking/tgt/default.nix`.

```nix
env.NIX_CFLAGS_COMPILE = toString [
	# Needed with GCC 12
	"-Wno-error=maybe-uninitialized"
];
```

mweinelt commented:
Well that was dumb.  I really need to start reading the actual error message.

The necessary change is

```patch
diff --git a/pkgs/applications/science/math/mxnet/default.nix b/pkgs/applications/science/math/mxnet/default.nix
index 1e7af371c50..d65de87d8eb 100644
--- a/pkgs/applications/science/math/mxnet/default.nix
+++ b/pkgs/applications/science/math/mxnet/default.nix
@@ -56,7 +56,6 @@ stdenv.mkDerivation rec {
 
   env.NIX_CFLAGS_COMPILE = toString [
     # Needed with GCC 12
-    "-Wno-error=maybe-uninitialized"
     "-Wno-error=uninitialized"
   ];

oddlama commented:
> The necessary change is [...]

Adjusted, I just can't test darwin myself unfortunately :D

mweinelt commented:
The change is needed on the native library, not the python library.

oddlama committed:
mxnet: fix darwin build

oddlama committed:
python310Packages.mxnet: add numpy >=1.24 compatibility patch

oddlama committed:
python310Packages.insightface: init at 0.7.3

mweinelt commented on pkgs/development/python-modules/mxnet/default.nix:
```diff
@@ -17,9 +17,17 @@ buildPythonPackage {
 
   LD_LIBRARY_PATH = lib.makeLibraryPath [ pkgs.mxnet ];
 
+  env.NIX_CFLAGS_COMPILE = toString [
+    # Needed with GCC 12
+    "-Wno-error=maybe-uninitialized"
+  ];
+
```
This can be removed.

oddlama commented:
Guess I should also start to read comments properly 😅. Added as suggested now.

mweinelt commented:
@ofborg build python311Packages.insightface

mweinelt commented:
Builds for me on both darwin systems.
